NEWS.md

```diff
@@ -1,6 +1,10 @@
 
 # renv 0.18.0  (UNRELEASED)
 
+* `renv` now attempts to infer remote dependencies for GitHub packages that
+  appear to be installed from sources; that is, for packages without a
+  recognizable remote source encoded in its DESCRIPTION file. (#841)
```

Comment:
```suggestion
* `renv` now attempts to infer remote dependencies for GitHub packages that
  appear to be installed from sources; that is, for packages without a
  recognizable remote source encoded in the DESCRIPTION file. (#841)
```

At first, I got confused about whose DESCRIPTION file we were talking about. It's not the DESCRIPTION of the dependency, but rather the DESCRIPTION of the renv project.

---

Just to review the basic facts in my use case:

The renv project is a Shiny app I'm deplying to Connect, which has no DESCRPTION or explicit version info of any kind. It also makes no explicit use of renv (but I know that the deployment machinery does).

The app makes use of several packages I have installed locally from source, such as gargle, which does declare GitHub as its source repo in the URL field of its DESCRIPTION file.

Reply:
Ah, ok, true. But I don't think many people realize that the installed DESCRIPTION gets modified in that way when you install from GitHub. I.e. info gets inserted that is not in the DESCRIPTION you see in the package source.

I think why I got confused is that I could also have solved my problem by using a DESCRIPTION file for my app, yeah? I could have put `Remotes: r-lib/gargle` there, which is what this PR is inferring.

Reply:
I don't think so; this PR is explicitly looking at the installed package's DESCRIPTION file, finding that it doesn't provide any metadata about its origin (ie: no fields like `RemoteType`), and so tries to infer that package should be installed from GitHub by its version number + presence of a GitHub URL in BugReports or URL.

Reply:
IOTW renv always uses what is _actually_ installed, where as the `DESCRIPTION` declares what should be installed.